JSDemos/Demos/List/ListSelection/React/App.js

```diff
@@ -14,82 +14,66 @@ const dataSource = new ArrayStore({
 const selectionModes = ['none', 'single', 'multiple', 'all'];
 const selectAllModes = ['page', 'allPages'];
 
-class App extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      selectionMode: 'all',
-      selectAllMode: 'page',
-      selectedItemKeys: [],
-    };
-    this.onSelectionModeChange = this.onSelectionModeChange.bind(this);
-    this.onSelectAllModeChange = this.onSelectAllModeChange.bind(this);
-    this.onSelectedItemKeysChange = this.onSelectedItemKeysChange.bind(this);
-  }
+export default function App() {
+  const [selectionMode, setSelectionMode] = React.useState('all');
+  const [selectAllMode, setSelectAllMode] = React.useState('page');
+  const [selectedItemKeys, setSelectedItemKeys] = React.useState([]);
 
-  onSelectionModeChange(args) {
-    this.setState({
-      selectionMode: args.value,
-    });
-  }
-
-  onSelectAllModeChange(args) {
-    this.setState({
-      selectAllMode: args.value,
-    });
-  }
-
-  onSelectedItemKeysChange(args) {
+  const onSelectedItemKeysChange = React.useCallback((args) => {
     if (args.name === 'selectedItemKeys') {
-      this.setState({
-        selectedItemKeys: args.value,
-      });
+      if (selectionMode !== 'none' || selectedItemKeys.length !== 0) {
+        setSelectedItemKeys(args.value);
+      }
     }
-  }
+  }, [selectionMode, selectedItemKeys, setSelectedItemKeys]);
```

Comment:
setSelectedItemKeys is not changed

Reply:
Yes I agree that React guarantees that setState function identity is stable and won’t change on re-renders and we can safe to omit from the useEffect or useCallback dependency list. 
Ok, let's trust him.

Reply:
Fixed in https://github.com/DevExpress/devextreme-demos/pull/2519/commits/4f7d234ff0d7038331e97b4f2fde6b2fd8be1d85